shell/platform/android/io/flutter/view/ResourceExtractor.java

```diff
@@ -48,19 +48,31 @@ static long getVersionCode(PackageInfo packageInfo) {
         }
     }
 
-    private class ExtractTask extends AsyncTask<Void, Void, Void> {
-        ExtractTask() { }
+    private static class ExtractTask extends AsyncTask<Void, Void, Void> {
+        private final String mDataDirPath;
+        private final HashSet<String> mResources;
+        private final AssetManager mAssetManager;
+        private final String mPackageName;
+        private final PackageManager mPackageManager;
+
+        ExtractTask(String dataDirPath, HashSet<String> resources, String packageName, PackageManager packageManager, AssetManager assetManager) {
```

Comment:
Would you mind add nullability annotations to all effected methods while you're here?

Reply:
I added them to the constructors and parameters - do you think they belong on any other methods here?

Reply:
The policy that I've recommended and been following is that every input parameter and every return type on every constructor and method should be annotated. I think at one point @mklim also recommended adding annotations to the member declarations as well, but I've been bad about remembering to do that.

Reply:
Done

Reply:
My reasoning is an extreme aversion to null. We're not enforcing non-null by default, so I've been recommending annotating all nullable members so internal code has less chance of trying to unbox something that's unset and hitting an NPE. We've also been annotating both `@Nullable` and `@NonNull` equally instead of assuming non-null and potentially falsely treating a null value as safe since the original code isn't really annotated. I think long term it would probably be worth enforcing non-null as some kind of statically linted default so we don't need to try and annotate so much here, but I don't know what would be involved setting that up.